conformance/system_macros/make_field.ion

```diff
@@ -0,0 +1,10 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Test Cases:
+// make_field can be invoked by name, address, qualified address, qualified name, binary address, and binary system e-expression op code.
+// the first argument must a non-null text value
+// annotations first argument are silently dropped
+// the second argument may be any expression
```

Comment:
Should we require that this be a singleton? Some thoughts:
* `make_field` suggests it makes a single name/value pair
* Accepting an expression that can expand to zero or 2+ values means each invocation requires an extra byte.

The extra byte might not matter much--I don't expect `make_field` will be called on the wire all that frequently. This is a situation where dropping the bitmap when there's only one variadic would be beneficial.

Reply:
Sure, and if you need to have any number of values, you can always create a template macro like this.
```
(macro make_fields (name vals*) (.for (val vals) (.make_field name val)))
```